rosbag2_storage/CMakeLists.txt

```diff
@@ -119,6 +124,14 @@ if(BUILD_TESTING)
     ${PROJECT_NAME}
     rosbag2_test_common::rosbag2_test_common
   )
-endif()
+
+  ament_add_gmock(test_qos
+    test/rosbag2_storage/test_qos.cpp)
+  target_link_libraries(test_qos
+    ${PROJECT_NAME}
+    rosbag2_test_common::rosbag2_test_common
+    yaml-cpp
```

Comment:
It might be that using `yaml-cpp` instead of `yaml-cpp-vendor` here causes the problem.

Reply:
See https://github.com/ros2/yaml_cpp_vendor/issues/10.
If you are ok I will try to update the CMakeLists.txt

Reply:
Update: I think that with `target_link_libraries` it is not possible to do so. However, in `yaml.hpp`, this was present:
```cpp
#ifdef _WIN32
// This is necessary because yaml-cpp does not always use dllimport/dllexport consistently
# pragma warning(push)
# pragma warning(disable:4251)
# pragma warning(disable:4275)
#endif
#include "yaml-cpp/yaml.h"
#ifdef _WIN32
# pragma warning(pop)
#endif
```
and I think this has to be done also in `qos.hpp`. Let's try :)

Reply:
Make sense to me. Thanks for addressing this issue.